File: Backend/auth-service/src/main/java/com/noc/authservice/service/RoleService.java
package com.noc.authservice.service;

import com.noc.authservice.dto.RoleRequest;
import com.noc.authservice.dto.RoleResponse;

import java.util.List;

public interface RoleService {

    RoleResponse createRole(RoleRequest request);

    List<RoleResponse> getRoles();

    RoleResponse getRoleById(Long id);

    RoleResponse updateRole(Long id, RoleRequest request);

    void deleteRole(Long id);
}

File: FRONTEND/src/app/core/services/role.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { RoleRequest, RoleResponse } from '../../shared/models';

@Injectable({ providedIn: 'root' })
export class RoleService {
  private readonly rolesUrl = `${environment.apiBaseUrl}/roles`;

  constructor(private readonly http: HttpClient) {}

  listRoles(): Observable<RoleResponse[]> {
    return this.http.get<RoleResponse[]>(this.rolesUrl);
  }

  createRole(request: RoleRequest): Observable<RoleResponse> {
    return this.http.post<RoleResponse>(this.rolesUrl, request);
  }

  updateRole(id: number, request: RoleRequest): Observable<RoleResponse> {
    return this.http.put<RoleResponse>(`${this.rolesUrl}/${id}`, request);
  }
}